### DISCUSSION TITLE
Custom Domain and subdomain support in codespaces

### DISCUSSION THREAD

**Discussion by sanjaydotpro** [community/community#45620] [Codespaces]
Hi,

My dev environment requires me to use docker-in-docker with codespaces to run applications like _traefik_, _authelia_, _nginx_ etc inside docker containers. 

Now for routing with _traefik_ and securing the app with _authelia_ requires me to specify subdomains for various containerised applications and there's seems to be **no way** to create subdomains or even use custom domains in codespaces.

Kindly let me know if there's any other way for me to setup dev environment with codespaces.

**Comment by hsein-bitar** [comment]
@MR-4O4 did you figure a way around this? Same situation here, I want the codespace to expose loopback domains other than localhost

**Comment by Rakeda** [comment]
Inside of the bash script just use the below command in the environment to update the /etc/hosts file for subdomain support.
`echo "127.0.0.1       sub.domain" | tee -a /etc/hosts`

**Comment by blauerberg** [comment]
I checked the creation log and it seems that the labels key is overwritten in compose.yml which is actually referenced in codespaces.

This can be seen in the files /workspaces/.codespaces/.persistedshare in the codespaces.

Since traefik references the labels key for routing, it appears that user-defined routing does not work.

However, I have not found any documentation on this specification.
